lib/active_record/connection_adapters/sqlserver_adapter.rb

```diff
@@ -152,8 +152,7 @@ def disable_referential_integrity
 
       def active?
         return false unless @connection
-        raw_connection_do 'SELECT 1'
-        true
+        raw_connection_do('SELECT 1') == 1
```

Comment:
IMO the previous one before the change is more robust.
Because the return value is independent of `raw_connection_do` implementation.

No matter how `raw_connection_do` changes in the future, as long as still raise expected exceptions.
This function will still work correctly.

Reply:
The problem with the previous version is that if raw_connection do does not return 1, but for example `false`, like it currently does when the connection is gone, no exception is raised which caused `active?` to return true, although the connection is not working.
With this change it will now only return true if the connection actually works, and false if an exception is raised, or anything else causes the connection to not return proper values any more.

Reply:
Additionally I'd like to point out that I did not get rid of the rescue block, so the previous behaviour of returning false if an connection_errors is raised is still the same.